01-resources-v2/branch-gen.yml

```diff
@@ -0,0 +1,39 @@
+resources:
+- name: atc
+  type: git
+  source: {uri: https://github.com/concourse/atc}
+  space: {branch: master}
+
+- name: atc-gen
+  type: git
+  source: {uri: "https://github.com/concourse/atc"}
+
+jobs:
+- name: gen
+  plan:
+  - get: atc
+    trigger: true
+  - task: gen
+    file: atc/ci/gen.yml
+    config:
+      platform: ...
+      image_resource: ...
+      outputs:
+      - name: generated-repo
+        # has the generated code committed to the repo
+      - name: branch-name
+        # has a 'name' file with `gen-(some deterministic hash)`
+  - put: atc-gen
+    params: {branch_name: branch-name/name, repository: gen-repo}
```

Comment:
Is `repository: gen-repo` supposed to be `atc-gen`?

Reply:
Nope, it should be `generated-repo` (the task output). Fixing.